### DISCUSSION TITLE
Task list checkboxes lack an accessible name

### DISCUSSION THREAD

**Discussion by patrickhlauke** [community/community#49126] [Accessibility]
### Select Topic Area

Bug

### Body

Once rendered as HTML, task list checkboxes currently lack an accessible name. Screen readers will just announce them as unlabelled checkboxes.

![A set of tasklist checkboxes, with one checkbox focused. NVDA's speech output next to it, showing that the focused checkbox is just announced as "checkbox not checked" without any label/name](https://user-images.githubusercontent.com/895831/222860679-6bf7fcbd-3d8d-47fc-ae89-8ba7728e3fb3.png)

Current markup of a rendered task list checkbox, simplified:

```html
<li class="…" data-title="Zipped and saved (PL)" data-position="…" draggable="false">
    <span class="handle">
        <svg class="…" aria-hidden="true" width="16" height="16">…</svg>
    </span>
    <input type="checkbox" id="" class="…"> Zipped and saved (PL)
    <button class=…" aria-label="Convert to issue" type="button" aria-describedby="button-description-3-4">
         …
    </button>
    <span hidden="" class="js-clear" id="button-description-3-4">…</span>
</li>
```

The simples, least disruptive fix to this would be to wrap the `<input>` and its label in an implicit `<label>`

```html
…
<label><input type="checkbox" id="" class="…"> Zipped and saved (PL)</label>
…
```

**Comment by patrickhlauke** [comment]
Noting here that this is still an issue.

Video recording using Chrome/NVDA tabbing through tasklist checkboxes - note the lack of announcement for the label

https://github.com/community/community/assets/895831/f93a4476-9cdc-4fee-a2de-8518bdcf51af

Current markup (same as before)

```
<input type="checkbox" id="" class="task-list-item-checkbox">
item one
<button class="…" aria-label="Convert to issue" type="button" aria-describedby="…">…</button>
```

**Comment by cehfisher** [comment]
Hello @patrickhlauke - thank you for providing this feedback and confirming it is still an issue. We will file this with our Accessibility team so that it can be reviewed and will follow-up here with any additional questions or updates. Thanks again!

**Comment by cehfisher** [comment]
Hello @patrickhlauke - This is a quick update to let you know the task list feature has recently been deprecated in favor of the new sub-issues functionality.

You can learn more about sub-issues here: [Introducing Sub-issues: Enhancing Issue Management on GitHub](https://github.blog/engineering/architecture-optimization/introducing-sub-issues-enhancing-issue-management-on-github)

We will close this issue as no longer relevant, but let us know if you have any questions or feedback on sub-issues. Thanks!
